src/video/SDL_shape.c

```diff
@@ -93,7 +93,11 @@ SDL_CalculateShapeBitmap(SDL_WindowShapeMode mode,SDL_Surface *shape,Uint8* bitm
                     pixel_value = *(Uint16*)pixel;
                     break;
                 case(3):
-                    pixel_value = *(Uint32*)pixel & (~shape->format->Amask);
+#if SDL_BYTEORDER == SDL_BIG_ENDIAN
+                    pixel_value = (pixel[0] << 16) | (pixel[1] << 8) | (pixel[2] << 0);
```

Comment:
pixel[0] is Uint8, so shifting left 16 may leave it 0.

Was this code tested?

Reply:
I did test this, albeit only on little-endian machines. I did try reversing the byteorder check and making sure the colours came out reversed, though, so it seemed like it was working both ways.

My suspicion is that, yes, ``pixel[0]`` is promoted to ``int``, so this should work on any system with ``sizeof(int)`` ≥ 3. I'm not opposed to adding additional casts to handle old 16-bit compilers if you want, though.

Reply:
Let's cast it explicitly so it's clear what's happening here.